src/filesystem/implementations/local.h

```diff
@@ -254,8 +256,12 @@ LocalFileSystem::MakeDirectory(const std::string& dir, const bool recursive)
   if (mkdir(dir.c_str(), S_IRWXU) == -1)
 #endif
   {
-    // Only allow the error due to parent directory does not exist
-    // if 'recursive' is requested
+    // Return success if directory already exists
+    if (errno == EEXIST) {
```

Comment:
Should it be a function argument on whether `file exist` is allowed?

Reply:
Good idea, I'll add it